api/app/GraphQL/Mutations/DisqualifyCandidate.php

```diff
@@ -20,6 +20,11 @@ public function __invoke($_, array $args)
 
         $candidate->pool_candidate_status = $reason;
         $candidate->final_decision_at = $now;
+
+        $finalDecision = $candidate->computeFinalDecision();
+        $candidate->computed_final_decision = $finalDecision['decision'];
+        $candidate->computed_final_decision_weight = $finalDecision['weight'];
```

Comment:
Is running this in the Mutations necessary? Won't this be a duplication of the work triggered by the Candidate Status Changed event?

Reply:
No, that event is only fires on `AssessmentStatusChanged` and `PoolCandidate::created`. Since it calls a `save` event I figured we may as well update it here so we don't need to make another trip to the database.

Reply:
Hmm. I'll be honest, I find it very confusing that updating the candidate status does not trigger the CandidateStatusChanged event. 😖 Unless there's some good reason not to, I recommend adding that to the PoolCandidateObserver.

Reply:
Moved to observer. I also now remeber why I moved it out. It was really bogging down but I realized the `save` was for some reason marking the status as changed and caused infiinte loop :face_with_spiral_eyes: 

Changing to `saveQuielty` seems to have fixed it :woman_shrugging: 

[compute final decision in observer](https://github.com/GCTC-NTGC/gc-digital-talent/pull/10920/commits/ba6baaedb8021be36774dd2560d07d4491afed04)